nixos/modules/system/boot/resolved.nix

```diff
@@ -11,24 +11,38 @@ let
 
   dnsmasqResolve = config.services.dnsmasq.enable && config.services.dnsmasq.resolveLocalQueries;
 
-  resolvedConf = ''
-    [Resolve]
-    ${optionalString (
-      config.networking.nameservers != [ ]
-    ) "DNS=${concatStringsSep " " config.networking.nameservers}"}
-    ${optionalString (cfg.fallbackDns != null) "FallbackDNS=${concatStringsSep " " cfg.fallbackDns}"}
-    ${optionalString (cfg.domains != [ ]) "Domains=${concatStringsSep " " cfg.domains}"}
-    LLMNR=${cfg.llmnr}
-    DNSSEC=${cfg.dnssec}
-    DNSOverTLS=${cfg.dnsovertls}
-    ${config.services.resolved.extraConfig}
-  '';
-
+  format = pkgs.formats.systemd { };
```

Comment:
Huh. TIL we have a `formats.systemd`, and I'm not sure it's right? Almost everywhere else, we use `settingsToSections`, e.g. in `logind`: https://github.com/NixOS/nixpkgs/blob/6a08e6bb4e46ff7fcbb53d409b253f6bad8a28ce/nixos/modules/system/boot/systemd/logind.nix#L64-L65

At the very least, this PR should also be using `settingsToSections`. Now, realistically, I suppose we should redo `formats.systemd` to not use the INI generator, since technically systemd format differs from INI in subtle ways. I'm not really sure though, and it seems out of scope for this PR. @LordGrimmauld, would like your thoughts on this though.